src/lib/common/include/sol-update.h

```diff
@@ -0,0 +1,185 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Intel Corporation nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "sol-str-slice.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @file
+ * @brief Routines to update a Soletta app. Provide way to check, fetch and
+ * install updates
+ */
+
+/**
+ * @defgroup Update Update
+ *
+ * Updating a Soletta based app may vary depending on Soletta environment and
+ * configuration. For instance, updating a static Linux Micro (PID 1) Soletta
+ * based app is only a matter of replacing app file, but when using shared libs,
+ * may involve update/install newer versions of used libs. So, Soletta provides
+ * different update modules to update Soletta based apps on different scenarios.
+ * Soletta will try to use the first module it can use, but if having different
+ * modules to be loaded, environment variable SOL_UPDATE_MODULE can be used to
+ * define a specific one, like SOL_UPDATE_MODULE=linux-micro-efi-update.
+ * When Soletta update module cheks an URL, it expects to get a JSON like
+ * @code{.js}
+ * {
+ *   "version":"1.0.0",
+ *   "url":"http://127.0.0.1/update",
```

Comment:
Please make this HTTPS-only (refuse to work with HTTP). Also check that the certificate is valid. If Soletta is built in debugging mode, might be a good idea for a key to bypass the certificate validity check check.